pantheon/src/main/java/tech/pegasys/pantheon/controller/NodeWhitelistController.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2018 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.controller;
+
+import tech.pegasys.pantheon.ethereum.permissioning.PermissioningConfiguration;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class NodeWhitelistController {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static List<String> nodeWhitelist;
```

Comment:
We might need a flag to know if the user set the property or not. If the user didn't, we won't do any permissioning.

An empty list can be ambiguous. It can indicate:

The user didn't whitelist any node (node whitelisting toggle on - don't talk to any node)
The user didn't specify the 'nodes-whitelist' property (node whitelisting toggled off)

We can assume that a null list means that the property hasn't been set (as opposed to an empty list when the user set the property without a value). But I think a flag is more semantic.